index.js

```diff
@@ -15,76 +16,82 @@ const defaultReqOptions = {
   userAgent: 'Schenkerianbot/1.0 (+https://github.com/mix/schenkerian)'
 }
 
-module.exports = function _export(options) {
-  let url = options.url
-  if (options.body) {
-    return analyze(url, options.body, options.returnSource)
-  } else {
-    if (options.tokens) _.merge(options, {jar: cookieJar(options.tokens, url)})
-    return requestAndSendToAnalyze(url, options)
+module.exports = function (options) {
+  const { url, tokens, body } = options
+  if (tokens) {
+    _.merge(options, {
+      jar: cookie.jar(tokens, url),
+      cookies: cookie.chromeCookies(tokens, url)
+    })
   }
-}
 
-function requestAndSendToAnalyze(url, options) {
-  let requestOptions = {
-    url: url,
-    timeout: options.timeout,
-    userAgent: options.userAgent,
-    fallbackRequest: options.fallbackRequest,
-    jar: options.jar
-  }
-  if (options.agent) {
-    requestOptions.agentClass = options.agent.agentClass
-    requestOptions.agentOptions = {
-      socksHost: options.agent.socksHost,
-      socksPort: options.agent.socksPort
-    }
+  if (body) {
+    return analyze(options)
   }
 
-  if (isMedia(url)) {
-    return requestPage(_.merge({
-      url
-    }, _.defaults(requestOptions, defaultReqOptions)))
-    .then(results => {
-      return {
-        url: results.url,
-        title: url,
-        image: url
+  return retrieveContent(url, options)
+}
+
+function retrieveContent(url, options) {
+  const { forceRequest, agent } = options
+  let requestOptions = _.defaults(
+    _.pick(options, ['url', 'timeout', 'userAgent', 'jar', 'cookies']),
+    defaultReqOptions
+  )
+
+  if (agent) {
+    const { agentClass, socksHost, socksPort } = agent
+    _.merge(requestOptions, {
+      agentClass,
+      agentOptions: {
+        socksHost,
+        socksPort
       }
     })
   }
 
-  if (options.forceRequest) {
+  if (isMedia(url) || forceRequest) {
     return requestPage(_.merge({
       url
-    }, _.defaults(requestOptions, defaultReqOptions)))
+    }, requestOptions))
     .then(results => {
-      return {
-        url: results.url,
+      // Media content can't be analyzed
+      // It does not contain html to be processed so just return the source
+      // Treat forceRequest flags the same way
+      return _.merge({
         title: url,
         image: url,
         source: results.body
-      }
+      }, _.pick(results, ['url']))
```

Comment:
Nitpick: Why `_.merge` and `_.pick` here instead of:
```javascript
return {
  title: url,
  image: url,
  url: results.url,
  source: results.body
}
```

Reply:
updated